### PR TITLE
Revamped About Page #1005

### LINKED ISSUE #1005: Revamp UI of About page and add our FB page link to it
Now that we have a somewhat active [Facebook page](https://www.facebook.com/WikimediaCommonsApp/) , it might be a good idea to add it to the app in a non-intrusive way so that users can follow for updates if they want to (also good for fostering a community around the app). The absolute most non-intrusive way would be to add it to the About activity, though that would be a bit obscure. If we can figure out a nice layout, adding it to the nav drawer instead (via the 'f' icon) could be an idea - though I am always open to opposition. ;)

This could be made a beginner-friendly task once we decide what exactly we want to do.

### PR DESCRIPTION
**Pull request by tshradheya**
Fixes #1005

**Solution**

Revamped the About page with required icons and links

Screenshot : 

layout-2018-01-20-010236

### CODE REVIEW

**Comment by maskaravivek** [comment]
Thanks for the PR @tshradheya. Please add a screenshot.

**Comment by tshradheya** [comment]
@maskaravivek Added in PR Description

File: app/src/main/res/layout/activity_about.xml [hunk 1]
```diff
@@ -62,6 +62,48 @@
...
+                android:layout_marginEnd="@dimen/activity_margin_horizontal"
+                >
+
+                <ImageView
+                    android:layout_width="@dimen/overflow_icon_dimen"
+                    android:layout_height="@dimen/overflow_icon_dimen"
+                    android:src="@drawable/ic_action_globe"
+                    android:contentDescription= "@string/commons_website"
+                    android:layout_margin="@dimen/activity_margin_horizontal"
+                    android:onClick="getOpenWebsiteIntent"
```
Comment (psh) [inline app/src/main/res/layout/activity_about.xml]: Binding `onClick` listeners in XML is error prone & easily breaks when refactoring;  **AboutActivity** is a much better place to do this - if you don't want to write the one-line to bind the listener yourself, you could annotate the `getOpenWebsiteIntent()` method with a [Butterknife](http://jakewharton.github.io/butterknife/) "@OnClick" annotation.

File: app/src/main/res/layout/activity_about.xml [hunk 2]
```diff
@@ -62,6 +62,48 @@
...
+                    />
+
+                <ImageView
+                    android:layout_width="@dimen/overflow_icon_dimen"
+                    android:layout_height="@dimen/overflow_icon_dimen"
+                    android:src="@drawable/ic_action_facebook"
+                    android:contentDescription= "@string/commons_facebook"
+                    android:layout_margin="@dimen/activity_margin_horizontal"
+                    android:onClick="getOpenFacebookIntent"
```
Comment (psh) [inline app/src/main/res/layout/activity_about.xml]: Binding `onClick` listeners in XML is error prone & easily breaks when refactoring;  **AboutActivity** is a much better place to do this - if you don't want to write the one-line to bind the listener yourself, you could annotate the `getOpenFacebookIntent()` method with a [Butterknife](http://jakewharton.github.io/butterknife/) "@OnClick" annotation.

File: app/src/main/res/layout/activity_about.xml [hunk 3]
```diff
@@ -62,6 +62,48 @@
...
+                    />
+
+                <ImageView
+                    android:layout_width="@dimen/overflow_icon_dimen"
+                    android:layout_height="@dimen/overflow_icon_dimen"
+                    android:src="@drawable/ic_action_github"
+                    android:contentDescription= "@string/commons_github"
+                    android:layout_margin="@dimen/activity_margin_horizontal"
+                    android:onClick="getOpenGithubIntent"
```
Comment (psh) [inline app/src/main/res/layout/activity_about.xml]: Binding `onClick` listeners in XML is error prone & easily breaks when refactoring;  **AboutActivity** is a much better place to do this - if you don't want to write the one-line to bind the listener yourself, you could annotate the `getOpenGithubIntent()` method with a [Butterknife](http://jakewharton.github.io/butterknife/) "@OnClick" annotation.

File: app/src/main/java/fr/free/nrw/commons/AboutActivity.java [hunk 4]
```diff
@@ -33,4 +36,29 @@ public void onCreate(Bundle savedInstanceState) {
         versionText.setText(BuildConfig.VERSION_NAME);
         initDrawer();
     }
+
+    public void getOpenFacebookIntent(View view) {
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/AboutActivity.java]: Strictly speaking, this isn't really a "get" method in the traditional Java sense as it's not returning anything.  Perhaps a name like "launchFacebook()" would be better?

File: app/src/main/java/fr/free/nrw/commons/AboutActivity.java [hunk 5]
```diff
@@ -33,4 +36,29 @@ public void onCreate(Bundle savedInstanceState) {
...
+            intent = new Intent(Intent.ACTION_VIEW, Uri.parse("fb://page/" + "1921335171459985"));
+            intent.setPackage("com.facebook.katana");
+            startActivity(intent);
+        } catch (Exception e) {
+            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.facebook.com/" + "1921335171459985")));
+        }
+
+    }
+
+    public void getOpenGithubIntent(View view) {
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/AboutActivity.java]: Strictly speaking, this isn't really a "get" method in the traditional Java sense as it's not returning anything.  Perhaps a name like "launchGithub()" would be better?

File: app/src/main/java/fr/free/nrw/commons/AboutActivity.java [hunk 6]
```diff
@@ -33,4 +36,29 @@ public void onCreate(Bundle savedInstanceState) {
...
+
+        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/commons-app/apps-android-commons\\"));
+        startActivity(browserIntent);
+    }
+
+    public void getOpenWebsiteIntent(View view) {
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/AboutActivity.java]: Strictly speaking, this isn't really a "get" method in the traditional Java sense as it's not returning anything.  Perhaps a name like "launchCommonsWebsite()" would be better?

**Comment by tshradheya** [comment]
Made requested changes.

Ready for Review

**Comment by neslihanturan** [comment]
Besides, I think resolution of .png icons are not correct. They looks pixelated. It would be great if we could use vectors instead of .pngs. Working with vectors would be easier to make them theme dependent too.

**Comment by tshradheya** [comment]
Okay. Thanks for the feedback.

Working on improvig it

**Comment by tshradheya** [comment]
Updated to vectors images and set theme dependency as well.

Ready for Review again

image

image

**Comment by neslihanturan** [comment]
@tshradheya thanks for this really fast fixes. This PR looks great except some minor things. One is it crashes on my 4.4.2 test device (not very minor actually:) ). I think it is compatibility issues with vector assets, I experienced this before but can't remember the solution. Maybe our some experienced developer remember the solution or you can solve it. 
The other is (really minor this time) facebook button looks a little big to me according to your screenshoots. I think thats it. After solving this issues, we can celebrate your first PR merged   :)

**Comment by tshradheya** [comment]
@neslihanturan I have started my work on the sizing and will make it perfect and equal for all three icons.

As for the crashing, I did a bit of researching and found out that the following needs to be added:

``` android {
   defaultConfig {
     vectorDrawables.useSupportLibrary = true
    }
 }
```

But I noticed that this has been already added and hence should work fine. This makes me wonder if the reason for crashing is actually the `vector assets`.

**Comment by neslihanturan** [comment]
@tshradheya removing app:srcCompat lines solves the issue as well. Let me find the solution while you work on the icons. I will post it here.

**Comment by neslihanturan** [comment]
I don't know why this solution works, but it works. Setting tint at layout file causes the problem. Carrying tint property to vector file solves the issue. As example:
```
<vector xmlns:android="http://schemas.android.com/apk/res/android"
        android:width="24dp"
        android:height="24dp"
        android:viewportWidth="24.0"
        android:viewportHeight="24.0"
        android:tint="?attr/iconsTheme">
    <path
        android:fillColor="#FF000000"
        android:pathData="M12,2C6.48...,5.39z"/>
</vector>
```
Besides you need to remove tint property from layout file to solve, of course. Can you add line breaks o vector files to make them more readible as in my code sample above? Can you also remove ic_action_globe.png images on your last commits (probably it is forgotten)?

**Comment by tshradheya** [comment]
Thanks a lot for the help. 

Made the necessary changes. Ready for review again

**Screenshot**

layout-2018-01-24-211356

**Comment by neslihanturan** [comment]
Everything looks correct to me, I will test and merge ASAP. Thanks for quick responses and great work @tshradheya !

**Pull request merged**